version: "3.8"
services:
  postgres:
    image: postgres:15.4-bullseye
    deploy:
      replicas: 1
      resources:
        limits:
          cpus: '2.0'
          memory: 1G
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
      placement:
        constraints: [ node.role == manager ]
    ports:
      - 5432:5432
    volumes:
       - postgres_data:/data/postgres
    networks:
      - shared_postgres
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      PGDATA: ${PGDATA}
  
  pgadmin:
    image: dpage/pgadmin4:7.5
    deploy:
      replicas: 1
      resources:
        limits:
          cpus: '2.0'
          memory: 1G
      restart_policy:
        condition: on-failure
        delay: 5s
        max_attempts: 3
        window: 120s
      placement:
        constraints: [ node.role == manager ]
    ports:
      - 5050:80
    volumes:
       - pgadmin_data:/var/lib/pgadmin
    networks:
      - shared_postgres
    environment:
      PGADMIN_DEFAULT_EMAIL: ${PGADMIN_DEFAULT_EMAIL}
      PGADMIN_DEFAULT_PASSWORD: ${PGADMIN_DEFAULT_PASSWORD}
      PGADMIN_CONFIG_SERVER_MODE: ${PGADMIN_CONFIG_SERVER_MODE}

networks:
  shared_postgres:
    external: true

volumes:
    postgres_data:
      external: true
    pgadmin_data:
      external: true
